processor/deltatocumulativeprocessor/internal/lineartelemetry/metrics.go

```diff
@@ -36,15 +37,15 @@ func New(set component.TelemetrySettings) (Metrics, error) {
 type Metrics struct {
 	metadata.TelemetryBuilder
 
-	tracked func() int
+	tracked *func() int
```

Comment:
fixes a bug discovered during testing where the callback update had no effect, so the metric was always 0

Reply:
Are you sure this is correct? `tracked func() int` is a function pointer already. `tracked *func() int` is a pointer to a pointer.

Reply:
in-fact yes.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e57fcf4154e591d8bcc14f5a00aa796efab9d279/processor/deltatocumulativeprocessor/internal/lineartelemetry/metrics.go#L20-L26

`trackedCb` captures `m.tracked` by value. if we later re-assign that `func()`, this is not reflected here.
by having it capture a pointer instead, we can update the underlying value and the closure correctly sees the new callback

Reply:
Would it make sense to use separate commits here? One that adds the test, even if it fails, another that fixes a bug.

If the there is a test failing before the fix it would be awesome since it show that it is indeed a bug :)

Reply:
sure

Reply:
Interesting.... I would have thought that go would have captured `m` and called its method. But it's capturing `m.tracked` directly?